smoke_tests/sqs_create_queue_test.go

```diff
@@ -0,0 +1,447 @@
+			"X-Amz-Target": "AmazonSQS.CreateQueue",
+		}).
+		WithJSON(cqr).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+
+	exp1 := models.CreateQueueResult{QueueUrl: fmt.Sprintf("%s/new-string-queue", af.BASE_URL)}
+
+	r1 := models.CreateQueueResult{}
+	json.Unmarshal([]byte(r), &r1)
+	assert.Equal(t, exp1, r1)
+
+	gqar := struct {
+		Action     string `xml:"Action"`
+		Attribute1 string `xml:"AttributeName.1"`
+		QueueUrl   string `xml:"QueueUrl"`
+	}{
+		Action:     "GetQueueAttributes",
+		Attribute1: "All",
+		QueueUrl:   fmt.Sprintf("%s/new-string-queue", af.BASE_URL),
+	}
+	r = e.POST("/").
+		WithForm(gqar).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+
+	dupe, _ := copystructure.Copy(sf.BASE_GET_QUEUE_ATTRIBUTES_RESPONSE)
+	exp3, _ := dupe.(app.GetQueueAttributesResponse)
+	exp3.Result.Attrs[0].Value = "30"
+	exp3.Result.Attrs[1].Value = "1"
+	exp3.Result.Attrs[7].Value = fmt.Sprintf("%s:new-string-queue", af.BASE_ARN)
+	exp3.Result.Attrs[8].Value = "{\"maxReceiveCount\": \"100\", \"deadLetterTargetArn\":\"new-queue-1\"}"
+	r3 := app.GetQueueAttributesResponse{}
+	xml.Unmarshal([]byte(r), &r3)
+	assert.Equal(t, exp3, r3)
+}
+
+func Test_CreateQueueV1_xml_no_attributes(t *testing.T) {
+	server := generateServer()
+	defer func() {
+		server.Close()
+		utils.ResetResources()
+	}()
+
+	utils.InitializeDecoders()
+
+	e := httpexpect.Default(t, server.URL)
+
+	r := e.POST("/").
+		WithForm(sf.CreateQueueV1RequestXML_NoAttributes).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+
+	exp1 := models.CreateQueueResponse{
+		Xmlns:    "http://queue.amazonaws.com/doc/2012-11-05/",
+		Result:   models.CreateQueueResult{QueueUrl: fmt.Sprintf("%s/new-queue-1", af.BASE_URL)},
+		Metadata: app.ResponseMetadata{RequestId: sf.REQUEST_ID},
+	}
+
+	r1 := models.CreateQueueResponse{}
+	xml.Unmarshal([]byte(r), &r1)
+	assert.Equal(t, exp1, r1)
+
+	r = e.POST("/").
+		WithForm(sf.ListQueuesRequestBodyXML).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+
+	exp2 := app.ListQueuesResponse{
+		Xmlns:    "http://queue.amazonaws.com/doc/2012-11-05/",
+		Result:   app.ListQueuesResult{QueueUrl: []string{fmt.Sprintf("%s/new-queue-1", af.BASE_URL)}},
+		Metadata: app.ResponseMetadata{RequestId: sf.REQUEST_ID},
+	}
+	r2 := app.ListQueuesResponse{}
+	xml.Unmarshal([]byte(r), &r2)
+	assert.Equal(t, exp2, r2)
+
+	r = e.POST("/").
+		WithForm(sf.GetQueueAttributesRequestBodyXML).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+
+	r3 := app.GetQueueAttributesResponse{}
+	xml.Unmarshal([]byte(r), &r3)
+	assert.Equal(t, sf.BASE_GET_QUEUE_ATTRIBUTES_RESPONSE, r3)
+}
+
+func Test_CreateQueueV1_xml_with_attributes(t *testing.T) {
+	server := generateServer()
+	defer func() {
+		server.Close()
+		utils.ResetResources()
+	}()
+
+	utils.InitializeDecoders()
+
+	e := httpexpect.Default(t, server.URL)
+
+	e.POST("/").
+		WithHeaders(map[string]string{
+			"Content-Type": "application/x-amz-json-1.0",
```

Comment:
The test name is `...xml...`, but the test request looks using `aws-json`.

Reply:
It seems that the test name and the protocol actually used for the other tests are different, so could you please review them?

Reply:
That is just to create a basic queue first so that I can set it as the Dead Letter Queue in the attributes below.  If that queue doesn't exist when I make the second request it will error.  But it's not particularly important in and of itself.  

Below, the next queue is created with XML which is the one with attributes populated - and the one I am asserting on at the end.